cty/value_ops.go

```diff
@@ -92,11 +96,6 @@ func (val Value) Equals(other Value) Value {
 		return UnknownVal(Bool)
 	}
 
-	// The types are equal, so the Nulls are equal.
```

Comment:
This new exception for null values creates a situation where two values of different types might become equal once they are known, so I think there's some further tweaking to do with the handling of unknown values.

I originally tried to express this in English but I think perhaps it's clearer to just express it as some additional test cases:

```go
		{
			NullVal(String),
			UnknownVal(Number),
			UnknownVal(Bool), // because second operand might eventually be null
		},
		{
			UnknownVal(String),
			NullVal(Number),
			UnknownVal(Bool), // because first operand might eventually be null
		},
		{
			UnknownVal(String),
			UnknownVal(Number),
			UnknownVal(Bool), // because both operands might eventually be null
		},
		{
			StringVal("hello"),
			UnknownVal(Number),
			False, // because no number value -- even null -- can be equal to a non-null string
		},
		{
			UnknownVal(String),
			NumberIntVal(1),
			False, // because no string value -- even null -- can be equal to a non-null number
		},
```

Reply:
OK, I see what you mean now. I think those unknown results are covered already, but I'll double check those last two.

Reply:
I think the first three of those cases would return `False` today if I read the logic correctly. If not, then great! :smile: